### ISSUE TITLE
i need to form a hybrid mL

### ISSUE THREAD

**Issue by fxgeniusllc-oss** [fxgeniusllc-oss/APEX-ARBITRAGE-SYSTEM#11]
i need to form this hybrid ml modelcontroller and integrate this file intp the buil;d tp further optimize and enhance speed and precision 

# ===================================================================
# OMNI-MEV-BOT — Dual Rust + LSTM Engine Edition
# Python AI Runtime: LSTM + ONNX Hybrid Controller
# ===================================================================

import os
import time
import json
import torch
import numpy as np
import asyncio
import redis
import requests
from fastapi import FastAPI, Request
from pydantic import BaseModel
from prometheus_client import start_http_server, Gauge
from torch import nn
from dotenv import load_dotenv
import onnxruntime as ort

# -------------------------------------------------------------------
# Load environment variables
# -------------------------------------------------------------------
load_dotenv()

LIVE_MODE = os.getenv("LIVE_TRADING", "false").lower() == "true"
AI_MODEL_PATH = os.getenv("AI_MODEL_PATH", "./models/lstm_omni.onnx")
AI_THRESHOLD = float(os.getenv("AI_THRESHOLD", 0.78))
REDIS_HOST = os.getenv("REDIS_HOST", "127.0.0.1")
REDIS_PORT = int(os.getenv("REDIS_PORT", 6379))

# -------------------------------------------------------------------
# Redis Cache + Prometheus Setup
# -------------------------------------------------------------------
redis_client = redis.Redis(host=REDIS_HOST, port=REDIS_PORT, decode_responses=True)
start_http_server(9090)

AI_PRED_CONFIDENCE = Gauge('ai_prediction_confidence', 'LSTM Prediction Confidence')
AI_LATENCY = Gauge('ai_inference_latency_ms', 'LSTM Inference Latency (ms)')

# -------------------------------------------------------------------
# Define the LSTM Model (for local inference fallback)
# -------------------------------------------------------------------
class OmniLSTM(nn.Module):
    def __init__(self, input_size=8, hidden_size=128, output_size=1, num_layers=2):
        super(OmniLSTM, self).__init__()
        self.lstm = nn.LSTM(input_size, hidden_size, num_layers, batch_first=True)
        self.fc = nn.Linear(hidden_size, output_size)

    def forward(self, x):
        out, _ = self.lstm(x)
        return torch.sigmoid(self.fc(out[:, -1, :]))

# -------------------------------------------------------------------
# Initialize ONNX Runtime + Torch Fallback
# -------------------------------------------------------------------
try:
    onnx_session = ort.InferenceSession(AI_MODEL_PATH)
    print(f"✅ ONNX model loaded: {AI_MODEL_PATH}")
except Exception as e:
    print(f"⚠️ ONNX model load failed, switching to PyTorch fallback: {e}")
    onnx_session = None
    torch_model = OmniLSTM()

# -------------------------------------------------------------------
# FastAPI Server for Inference Requests
# -------------------------------------------------------------------
app = FastAPI(title="OMNI-MEV AI Engine", version="2.0")

class PredictionRequest(BaseModel):
    features: list

@app.post("/predict")
def predict(req: PredictionRequest):
    start = time.time()

    # Input preparation
    features = np.array(req.features, dtype=np.float32)
    features = np.expand_dims(features, axis=(0, 1))  # (batch, seq, features)

    # ----------------------------------------------------------------
    # Inference logic
    # ----------------------------------------------------------------
    try:
        if onnx_session:
            result = onnx_session.run(None, {onnx_session.get_inputs()[0].name: features})[0]
            confidence = float(result[0][0])
        else:
            with torch.no_grad():
                tensor_in = torch.tensor(features)
                confidence = float(torch_model(tensor_in).numpy()[0][0])

        AI_PRED_CONFIDENCE.set(confidence)
        AI_LATENCY.set((time.time() - start) * 1000)

        redis_client.set("ai:last_confidence", confidence)

        decision = confidence > AI_THRESHOLD
        payload = {
            "decision": bool(decision),
            "confidence": confidence,
            "threshold": AI_THRESHOLD,
            "mode": "LIVE" if LIVE_MODE else "SIMULATION",
        }

        # Optional: notify Rust engine for live action
        if LIVE_MODE and decision:
            try:
                requests.post("http://localhost:7000/execute", json=payload, timeout=2)
            except Exception:
                pass

        return payload

    except Exception as e:
        return {"error": str(e)}

@app.get("/status")
def status():
    return {"ai_engine": "online", "mode": "LIVE" if LIVE_MODE else "SIMULATION"}

# -------------------------------------------------------------------
# Background: Live Streaming Loop
# -------------------------------------------------------------------
async def ai_loop():
    while True:
        try:
            last_conf = redis_client.get("ai:last_confidence")
            print(f"📡 Confidence: {last_conf}")
            await asyncio.sleep(5)
        except Exception:
            await asyncio.sleep(10)

# -------------------------------------------------------------------
# Entry Point
# -------------------------------------------------------------------
if __name__ == "__main__":
    print(f"⚙️ Starting AI Engine (Mode: {'LIVE' if LIVE_MODE else 'SIMULATION'})")
    loop = asyncio.get_event_loop()
    loop.create_task(ai_loop())
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8001)
